examples/webserver/threadpool/src/main/java/io/helidon/examples/webserver/threadpool/GreetService.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.examples.webserver.threadpool;
+
+import java.util.Collections;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.json.Json;
+import javax.json.JsonBuilderFactory;
+import javax.json.JsonException;
+import javax.json.JsonObject;
+
+import io.helidon.common.configurable.ThreadPoolSupplier;
+import io.helidon.common.context.ContextAwareExecutorService;
+import io.helidon.common.context.Contexts;
+import io.helidon.common.http.Http;
+import io.helidon.config.Config;
+import io.helidon.webserver.Routing;
+import io.helidon.webserver.ServerRequest;
+import io.helidon.webserver.ServerResponse;
+import io.helidon.webserver.Service;
+
+/**
+ * A simple service to greet you. Examples:
+ *
+ * Get default greeting message:
+ * curl -X GET http://localhost:8080/greet
+ *
+ * Get greeting message for Joe:
+ * curl -X GET http://localhost:8080/greet/Joe
+ *
+ * Change greeting
+ * curl -X PUT -H "Content-Type: application/json" -d '{"greeting" : "Howdy"}' http://localhost:8080/greet/greeting
+ *
+ * The message is returned as a JSON object
+ */
+
+public class GreetService implements Service {
+
+    /**
+     * The config value for the key {@code greeting}.
+     */
+    private final AtomicReference<String> greeting = new AtomicReference<>();
+
+    private static final JsonBuilderFactory JSON = Json.createBuilderFactory(Collections.emptyMap());
+
+    private static final Logger LOGGER = Logger.getLogger(GreetService.class.getName());
+
+    private static ExecutorService myThreadPool;
+
+    GreetService(Config config) {
+        greeting.set(config.get("app.greeting").asString().orElse("Ciao"));
+
+        // Build a thread pool using the configuration
+        myThreadPool = ThreadPoolSupplier.builder().config(config.get("application-thread-pool")).build().get();
+        if (myThreadPool instanceof ContextAwareExecutorService) {
```

Comment:
1. This is always true
2. Why would we want to do this? 
3. If we want to show how to use a custom executor, we may want to create an executor service (such as using `Executors....` and then wrap it using `Contexts.wrap(ExecutorService)` rather than creating a Helidon executor service and unwrapping it

Reply:
The only reason I'm unwrapping is to log the configuration of the threadpool so that the user can see configuration changes reflected in the threadpool settings.  I think this would be a common question from users: how do I confirm that the threadpool is actually using the configuration I'm passing (to make sure I haven't made a typo in my config).